src/dash-table/dash/LazyFactory.ts

```diff
@@ -0,0 +1,39 @@
+import { lazy } from 'react';
+
+type Boolean = Promise<boolean> | boolean;
+type LazyComponent = React.LazyExoticComponent<React.ComponentType<any>>;
+
+interface IResult {
+    isReady: Boolean;
+    get: LazyComponent;
+}
+
+export default (
+    target: () => JSX.Element,
+    promise: () => Promise<{ default: React.ComponentType<any> }>
+): LazyComponent => {
+    let resolve: (value: Boolean) => void;
+    const isReady = new Promise<boolean>(r => {
+        resolve = r;
+    });
+
+    const state: IResult = {
+        isReady,
+        get: lazy(async () => {
+            const res = await promise();
+
+            setTimeout(async () => {
+                await resolve(true);
+                state.isReady = true;
+            }, 0);
+
+            return res;
+        })
+    };
+
+    Object.defineProperty(target, '_dashprivate_isLazyComponentReady', {
+        get: () => state.isReady
+    });
```

Comment:
Wraps a promise inside a React.lazy -- when it resolves the async component has loaded, a 2nd promise is resolved and flag set to `true` for the renderer to know when to initiate/block callbacks in https://github.com/plotly/dash/pull/899/files#diff-15692ba768eeab56e95aa409ab7c19b7R262